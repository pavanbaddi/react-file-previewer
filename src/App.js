import React, {useState} from 'react';
import './App.css';
import FileExplorer from './PageComponents/FileExplorer';

function App() {

  const [files, setFiles] = useState([])
  const [prevFile, setPrevFile] = useState(null)

  const onHandleFileChanges = ( event ) => {
    const element = event.target
    const tempFiles = []
    
    if(element.files.length){
      for(let file of element.files){
        tempFiles.push({file, url: URL.createObjectURL(file)})
      }
    }
    setFiles(tempFiles)
    element.value = "";
  }

  const prevChoosenFile = (file) => {
    setPrevFile(file)
  }


  return (
    <div>
      <div className='form-container' >
        <div>
          <label>File Uploader</label>
        </div>
        <input data-testid="file-input" type='file' onChange={onHandleFileChanges} multiple/>
      </div>


      <FileExplorer files={files} prevFile={prevFile} prevChoosenFile={prevChoosenFile} />

    </div>
  );
}

export default App;
