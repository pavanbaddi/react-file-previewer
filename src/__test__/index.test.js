import React from "react";
import {render, screen} from "@testing-library/react";
import user from "@testing-library/user-event";
import PdfContextProvider from "../Contexts/PdfContext";
import App from "../App";
import pdfbase64 from "./utils/pdfbase64";
import {act} from "react-dom/test-utils"
import ReactDOM from "react-dom/client"

const Component = ( {children} ) => {
    return (
        <React.StrictMode>
            <PdfContextProvider>
                {children}
            </PdfContextProvider>
        </React.StrictMode>
    )
}


it("must render", () => {
    render(<Component ><App/></Component>)
    const fileInput = screen.getByTestId("file-input");
    expect(fileInput).toBeInTheDocument()
})

it("before uploading any file", () => {
    render(<Component ><App/></Component>)
    let listing = null;
    
    try {
        listing = screen.getByTestId("listing")
    } catch (error) {
        
    }
    
    expect(listing).not.toBeInTheDocument()
})


let container = null;

beforeEach(() => {
    container = document.createElement("div")
    document.body.appendChild(container)
})

afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });

it("upload a pdf file", () => {
    global.URL.createObjectURL = jest.fn((file) => "")

    act(() => {
        ReactDOM.createRoot(container).render(<Component ><App/></Component>)
    });

    const fileInput = screen.getByTestId("file-input");

    const blobOptions= {type: "application/pdf"};
    const pdfBlob = new Blob([pdfbase64], blobOptions )
    const pdfFile = new File([pdfBlob], "test-pdf.pdf", {type: "application/pdf"})
    File.prototype.text = jest.fn().mockResolvedValueOnce(pdfbase64)
    
    act(() => {
        user.upload(fileInput, pdfFile)
    });

    const listing = screen.getByTestId("listing")
    
    expect(listing.childElementCount).toEqual(1);
})
