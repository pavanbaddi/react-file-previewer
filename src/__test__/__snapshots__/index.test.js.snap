// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1 1`] = `
<div>
  <div>
    <div
      class="form-container"
    >
      <div>
        <label>
          File Uploader
        </label>
      </div>
      <input
        data-testid="file-input"
        multiple=""
        type="file"
      />
    </div>
  </div>
</div>
`;

exports[`test upload journey upload a pdf file 1`] = `
<div>
  <div>
    <div
      class="form-container"
    >
      <div>
        <label>
          File Uploader
        </label>
      </div>
      <input
        data-testid="file-input"
        multiple=""
        type="file"
      />
    </div>
    <div
      class="file-viewer-container"
    >
      <div
        class="listing"
      >
        <h4>
          List of Selected Files
        </h4>
        <ul
          data-testid="listing"
        >
          <li
            class=" active "
          >
            <p>
              test-pdf.pdf
               (Click to View) 
            </p>
          </li>
          <li
            class="  "
          >
            <p>
              test-pdf.pdf
               (Click to View) 
            </p>
          </li>
        </ul>
      </div>
      <div
        class="viewer"
        data-testid="viewer"
      >
        <div>
          <div>
            <h4>
              Options
            </h4>
          </div>
          <button
            type="button"
          >
             
            Enable
             Thumbnail
          </button>
          <button
            type="button"
          >
             
            Enable
             Rotation
          </button>
          <button
            type="button"
          >
             
            Disable
             Default Layout
          </button>
          <button
            type="button"
          >
             
            Disable
             Reading Progress
          </button>
          <button
            type="button"
          >
             
            Disable
             Download
          </button>
          <button
            type="button"
          >
            Print
          </button>
          <button
            type="button"
          >
             
            Disable
             Zoom Level
          </button>
          <div>
            <h5>
              Zoom Options
            </h5>
            <select>
              <option
                value="ActualSize"
              >
                Actual Size
              </option>
              <option
                value="PageFit"
              >
                Page Fit
              </option>
              <option
                value="PageWidth"
              >
                Page Width
              </option>
              <option
                value="0.5"
              >
                50%
              </option>
              <option
                value="1"
              >
                100%
              </option>
              <option
                value="2"
              >
                200%
              </option>
            </select>
          </div>
        </div>
        <button
          style="background-color: rgb(53, 126, 221); border-radius: 4px; color: rgb(255, 255, 255); cursor: pointer; padding: 8px;"
        >
          Download
        </button>
        <div
          class="rpv-core__viewer rpv-core__viewer--light"
          data-testid="core__viewer"
          style="height: 100%; width: 100%;"
        />
      </div>
    </div>
  </div>
</div>
`;
